extension/healthcheckextension/healthcheckextension.go

```diff
@@ -109,13 +109,34 @@ func (hc *healthCheckExtension) Start(_ context.Context, host component.Host) er
 	return nil
 }
 
+// base handler function
+func (hc *healthCheckExtension) baseHandler() http.Handler {
+	if hc.config.StaticResponseBody != "" {
+		return http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+			if hc.state.Get() == healthcheck.Ready {
+				w.WriteHeader(http.StatusOK)
+				if hc.config.StaticResponseBody != "" {
+					w.Write([]byte(hc.config.StaticResponseBody))
+				}
+			} else {
+				w.WriteHeader(http.StatusServiceUnavailable)
```

Comment:
would it make sense to have a static response body on error too?

Reply:
I wondered the same.

In my case it wasn’t needed because the logic just check if the status is 200 and if the body contains a specific string.
If either are false, it fails the check.

If preferred, I can modify it to something a little more refined like:
```
static_response_code:
  enabled: true/false
  ready: ""
  not_ready: ""
```

Reply:
I think you can get away with:
```
response_body:
  healthy: ""
  unhealthy: ""
```

No need for enabled, and I renamed the fields:
* static_response_code -> response_body as it is static, anyway, and it's the response body contents
* ready -> healthy since I guess the status can flip back and forth, not just on readiness.

Reply:
The enabled flag was just to clarify that if either are provided, it would override both responses. 
I’ll update the PR with your suggestion